### PR TITLE
Add a check for empty arrays

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
@evanmwillhite lets talk about this. It doesn't solve the issue I'm having 100%, but it does get 50% of the way.

┆Issue is synchronized with this [Clickup task](https://app.clickup.com/t/86a2fe314) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by mirzu** [comment]
Evan to Add more details and assign to the right developer

**Comment by evanmwillhite** [comment]
This might need to be done in https://github.com/emulsify-ds/emulsify_twig (PHP) and https://github.com/emulsify-ds/emulsify-twig-extensions (JS)

File: packages/bem-twig-extension/lib/bem-twig-extension.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ function bemTwigExtension(Twig) {
     }
     
     // If extra non-BEM classes are added.
-    if (extra.length && Array.isArray(extra)) {
+    if (extra.length && Array.isArray(extra) && extra != '') {
```
Comment (elliotttf) [inline packages/bem-twig-extension/lib/bem-twig-extension.js:34]: You might be able to get what you want by just checking to see if extra is an array:

```javascript
if (Array.isArray(extra)) {
  classes = [
    ...classes,
    ...extra,
  ];
}
```


Also, hi friends, apparently I'm subscribed to this repo 😂

**Comment by evanmwillhite** [comment]
Ready for review here https://github.com/emulsify-ds/emulsify_twig/pull/14

**Comment by ModulesUnraveled** [comment]
@ryanhagerty mentioned that this only seems to be an issue in Drupal, not storybook. So we suspect the issue is actually in this repo: https://github.com/emulsify-ds/emulsify_twig

**Comment by callinmullaney** [comment]
@codechefmarc Lets get this one across the finish line. Happy to review if you need to make additional changes to what is already provided.

**Comment by codechefmarc** [comment]
@callinmullaney - finally looking into this one, sorry for the delay. In reviewing the changes here, it seems this additional check does remove an added space in the DOM when there is an empty string in as an array item:

`<ul {{ bem('list', ['modifier'], status__base_class, [''])}}>` produces:

`<ul class="status__list status__list--modifier ">` vs `<ul class="status__list status__list--modifier">` with the fix.

As it currently stands, I don't think this is a problem as the browsers will ignore any extra spaces that are added to the end of a class list like this. However, it does make it cleaner with this fix. I'm fine with adding this.

**Review by callinmullaney — APPROVED** [review]

**Comment by callinmullaney** [comment]
@codechefmarc Confirmed your testing and think this is good to merge. Feel free to do so and cut a release when you have some time.

**Pull request merged**